### PR TITLE
feature/webfixes fix #186

### LINKED ISSUE #186: Team View: News are empty
Not seeing any news (`Posts` with `verb == 'announce'`) despite the `feed` showing that there are some:
image 
image

They should show up in the `news`-section, too.

### PR DESCRIPTION
**Pull request by davemecha**
Fix issue #186 and show correct content in news and feed

Also GitPod support is added (with small problems). Should not interfere with local development.

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -17,7 +17,7 @@ services:
       - mongo
   dashboard:
     image: parseplatform/parse-dashboard
-    command: --dev --appId APPLICATION_ID --masterKey MASTER_KEY --serverURL "http://localhost:1337/parse" --appName affinity
+    command: --dev --appId APPLICATION_ID --masterKey MASTER_KEY --serverURL VUE_APP_PARSE_URL --appName affinity
```
Comment (gnunicorn) [inline docker-compose.yml:20]: I don't think this works. this is not using a variable...

File: .gitpod.Dockerfile [hunk 2]
```diff
@@ -0,0 +1,14 @@
+FROM gitpod/workspace-full
```
Comment (gnunicorn) [inline .gitpod.Dockerfile:1]: Can we move all .gitpod.* things not necessary in the root into a subfolder please?

File: .gitpod.yml [hunk 3]
```diff
@@ -0,0 +1,7 @@
+image:
```
Comment (gnunicorn) [inline .gitpod.yml:1]: Thinking about this further, let's split this into a separate PRs, please. We generally do atomic PRs with a particular or at least related changes - these are unrelated and should  probably be considered independent from another.

File: src/components/render-objects.vue [hunk 4]
```diff
@@ -54,7 +54,8 @@ export default defineComponent({
     objects: {
       type: Array,
       required: false
-    }
+    },
+    class: String,
```
Comment (gnunicorn) [inline src/components/render-objects.vue:58]: what's this about?

File: src/utils/setup.ts [hunk 5]
```diff
@@ -70,32 +70,36 @@ export function initInstallation(): Promise<Parse.Installation> {
   }
   const promise: Promise<Parse.Installation> = new Promise(
     (resolve, reject) => {
-      PushNotifications.addListener("registration", (token: Token) => {
-        generateInstallation({ deviceToken: token.value })
-          .then((x: Parse.Installation) => resolve(x))
-          .catch((x: any) => reject(x));
-      });
+      if (Capacitor.isPluginAvailable("PushNotifications")) {
```
Comment (gnunicorn) [inline src/utils/setup.ts:73]: Description should probably mentioned this fix being in this PR,  too.

File: src/views/Team/Outlet.vue [hunk 6]
```diff
@@ -329,30 +339,30 @@ export default defineComponent({
...
+        ].toPointer();
+
+        await Promise.all([
+          this.store.dispatch("teams/fetchNews", teamPointer),
+          this.store.dispatch("teams/fetchActivities", teamPointer),
+          this.store.dispatch("teams/fetchSubteams", teamPointer),
+        ]);
+
+        this.loading = false;
+      } catch (error) {
```
Comment (gnunicorn) [inline src/views/Team/Outlet.vue:362]: what's this general `try-catch` about? unresolved/broken promises that aren't caught are shown in the log anyways, and so are any other execution problems...

File: src/views/Team/Outlet.vue [hunk 7]
```diff
@@ -329,30 +339,30 @@ export default defineComponent({
...
-            this.store.dispatch("teams/fetchNews", teamPointer),
-            this.store.dispatch("teams/fetchSubteams", teamPointer),
-          ]);
-        })
-        .then(() => {
-          this.loading = false;
-        });
+        if (!this.store.getters.teamsBySlug[slug])
+          await this.store.dispatch("teams/fetch", slug);
+
```
Comment (gnunicorn) [inline src/views/Team/Outlet.vue:350]: Style..

```suggestion
        if (!this.store.getters.teamsBySlug[slug]) {
          await this.store.dispatch("teams/fetch", slug);
        }
```

**Review by gnunicorn — CHANGES_REQUESTED** [review]
Nice!!

Let's split this into separate atomic PRs, please. Also minor fixes needed to improve code style and let's add tests, just to make sure future changes don't introduce regressions... iirc, we already have tests around the teamview, shouldn't be complicated to add any.

File: src/utils/setup.ts (hunk 5 above)
Comment (davemecha) [reply src/utils/setup.ts:73]: separate PR https://github.com/TeamFranka/affinity/pull/190

File: docker-compose.yml (hunk 1 above)
Comment (davemecha) [reply docker-compose.yml:20]: fixed by @gnunicorn in https://github.com/TeamFranka/affinity/commit/fc54ca4b8bab645939860b80d8e85ade03aa539f

File: .gitpod.Dockerfile (hunk 2 above)
Comment (davemecha) [reply .gitpod.Dockerfile:1]: fixed by @gnunicorn in https://github.com/TeamFranka/affinity/commit/c6abefd546d703dd1bdc0af9e78839776e9170d3#diff-050265bb25fc16211ce409bd85cd699db1be9296267f976eb0f3cd85c5ab7ff9

File: .gitpod.yml (hunk 3 above)
Comment (davemecha) [reply .gitpod.yml:1]: fixed by @gnunicorn in https://github.com/TeamFranka/affinity/commit/c6abefd546d703dd1bdc0af9e78839776e9170d3

File: src/components/render-objects.vue (hunk 4 above)
Comment (davemecha) [reply src/components/render-objects.vue:58]: new PR for that: https://github.com/TeamFranka/affinity/pull/194

File: src/utils/setup.ts (hunk 5 above)
Comment (davemecha) [reply src/utils/setup.ts:73]: fixed by https://github.com/TeamFranka/affinity/pull/190

File: src/views/Team/Outlet.vue (hunk 6 above)
Comment (davemecha) [reply src/views/Team/Outlet.vue:362]: https://github.com/TeamFranka/affinity/pull/195

File: src/views/Team/Outlet.vue (hunk 7 above)
Comment (davemecha) [reply src/views/Team/Outlet.vue:350]: https://github.com/TeamFranka/affinity/pull/196
